src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -156,6 +156,17 @@ internal MetricPointsAccessor GetMetricPoints()
             return new MetricPointsAccessor(this.metricPoints, this.currentMetricPointBatch, this.batchSize, this.startTimeExclusive, this.endTimeInclusive);
         }
 
+        internal MetricPointsAccessor GetDeepCloneMetricPoints()
+        {
+            var deepClonedMetricPoints = new MetricPoint[this.metricPoints.Length];
+            for (int i = 0; i < this.metricPoints.Length; i++)
+            {
+                deepClonedMetricPoints[i] = this.metricPoints[i].DeepCopy();
```

Comment:
Just the assignment operator should copy the `MetricPoint` struct.

If the metric `AggregationType` is `Histogram`, we need to additionally copy `HistogramBuckets.SnapshotBucketCounts` for each `MetricPoint`.

Reply:
The assignment would copy the `struct MetricPoint` but will not copy the `class HistogramBuckets`. 
I don't see a way to selectively copy only one field of `HistogramBuckets`. 
I'm open to suggestions.

Reply:
How about something like this:

```
            var clonedMetricPoints = new MetricPoint[this.metricPoints.Length];
            for (int i = 0; i < this.metricPoints.Length; i++)
            {
                clonedMetricPoints[i] = this.metricPoints[i];
                for (int j = 0; j < this.metricPoints[i].HistogramBuckets.SnapshotBucketCounts.Length; j++)
                {
                    clonedMetricPoints[i].HistogramBuckets.SnapshotBucketCounts[j] = this.metricPoints[i].HistogramBuckets.SnapshotBucketCounts[j];
                }
            }
```

Reply:
Am i missing something?

1. `clonedMetricPoints[i] = this.metricPoints[i];`
    This will create a deep copy of the struct. aka `MP2 = MP1`.
    But these two structs will have references to the same instance of HistogramBuckets.

2. `clonedMetricPoints[i].HistogramBuckets.SnapshotBucketCounts[j] = this.metricPoints[i].HistogramBuckets.SnapshotBucketCounts[j];`
    This is working with the same instance of HistogramBuckets. aka `MP2.HB1.SBC1 = MP1.HB1.SBC1`

Reply:
I'm sorry about the confusion. What I suggested won't work. 

I think we need to have duplicate storage for both `MetricPoint[]` and each `MetricPoint`'s `SnapshotBucketCounts`. Also, we need to understand what's the best place to persist the additional storage. I don't think this would be as straightforward.

Reply:
I think we first need to decide on the usage of `InMemoryExporter` for Metrics. If we decide to clear the collection of exported items on every export call we don't have to deep copy anything. We would just have the user inspect the collection of exported items after every export.